components/device-mgt/org.wso2.carbon.device.mgt.core/src/test/java/org/wso2/carbon/device/mgt/core/search/ProcessorImplTest.java

```diff
@@ -0,0 +1,174 @@
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * This class holds unit test cases for org.wso2.carbon.device.mgt.core.search.mgt.impl.ProcessorImpl
+ * */
+public class ProcessorImplTest extends BaseDeviceManagementTest{
+
+    private DeviceAccessAuthorizationService deviceAccessAuthorizationService;
+    private static final Log log = LogFactory.getLog(SearchManagementServiceTest.class);
+    private static List<DeviceIdentifier> deviceIdentifiers = new ArrayList<>();
+    private static final String DEVICE_ID_PREFIX = "SEARCH-DEVICE-ID-";
+    private static final String DEVICE_TYPE = "SEARCH_TYPE";
+
+    @BeforeClass
+    public void init() throws Exception {
+        deviceAccessAuthorizationService = DeviceManagementDataHolder.getInstance()
+                .getDeviceAccessAuthorizationService();
+
+        for (int i = 0; i < 5; i++) {
+            deviceIdentifiers.add(new DeviceIdentifier(DEVICE_ID_PREFIX + i, DEVICE_TYPE));
+        }
+        DeviceManagementProviderService deviceMgtService = new DeviceManagementProviderServiceImpl();
+        DeviceManagementServiceComponent.notifyStartupListeners();
+        DeviceManagementDataHolder.getInstance().setDeviceManagementProvider(deviceMgtService);
+        deviceMgtService.registerDeviceType(new TestDeviceManagementService(DEVICE_TYPE,
+                MultitenantConstants.SUPER_TENANT_DOMAIN_NAME));
+
+        List<Device> devices = TestDataHolder.generateDummyDeviceData(deviceIdentifiers);
+        for (Device device : devices) {
+            device.setDeviceInfo(Utils.getDeviceInfo());
+            deviceMgtService.enrollDevice(device);
+        }
+        List<Device> returnedDevices = deviceMgtService.getAllDevices(DEVICE_TYPE, true);
+        for (Device device : returnedDevices) {
+            if (!device.getDeviceIdentifier().startsWith(DEVICE_ID_PREFIX)) {
+                throw new Exception("Incorrect device with ID - " + device.getDeviceIdentifier() + " returned!");
+            }
+        }
+    }
+
+    @Test(description = "Test the Search Processor")
+    public void testWithNoDeviceAccessAuthorization() throws Exception {
+        SearchContext context = new SearchContext();
+        List<Condition> conditions = new ArrayList<>();
+
+        Condition cond = new Condition();
+        cond.setKey("batteryLevel");
+        cond.setOperator("=");
+        cond.setValue("40");
+        cond.setState(Condition.State.AND);
+        conditions.add(cond);
+
+        context.setConditions(conditions);
+
+        ProcessorImpl processor = new ProcessorImpl();
+        Field deviceAccessAuthorizationServiceField = ProcessorImpl.class.getDeclaredField
+                ("deviceAccessAuthorizationService");
+        deviceAccessAuthorizationServiceField.setAccessible(true);
+        deviceAccessAuthorizationServiceField.set(processor, null);
+
+        List<Device> searchedDevices = processor.execute(context);
+
```

Comment:
Remove unwanted spaces. Correct them in all the palces

Reply:
This is fixed in https://github.com/wso2/carbon-device-mgt/pull/1082/commits/e70eefe74b4486a1f41f521ee517076951722de4